### ISSUE TITLE
bootstrap: If password or key file are required we should tell the flags to do so

### ISSUE THREAD

**Issue by tas50** [chef/chef#8481]
If you try to bootstrap a node using a username in the URI you will receive a train error instead of a friendly knife error. The user shouldn't see train errors since they don't need to know what handles the transport under the hood. Additionally if there are flags missing we should make sure to let people know what flags they can use so they don't have to hit help or the docs site on their own.

```
bundle exec knife bootstrap ssh://ubuntu@ec2-34-221-181-2.us-west-2.compute.amazonaws.com --channel unstable -N tim_test
Creating new client for tim_test
Creating new node for tim_test
Connecting to ec2-34-221-181-2.us-west-2.compute.amazonaws.com
ERROR: Train::ClientError: Your SSH Agent has no keys added, and you have not specified a password or a key file
```

Also: Knife call it an identity file vs. train calling it a key file, which further confuses the issue

**Comment by vsingh-msys** [comment]
Address in https://github.com/chef/chef/pull/8856

**Comment by vsingh-msys** [comment]
Fixes #8856 merged in master
